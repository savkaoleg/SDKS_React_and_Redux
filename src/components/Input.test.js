import React from 'react'
import expect from 'expect'
import { shallow } from 'enzyme'
import Input from './Input'

describe('Input', () => {
  it('Renders correctly', () => {
    const component = shallow(<Input />)

    expect(component).toMatchSnapshot()
  })

  it('Value', () => {
    const testString = 'abcde'
    const component = shallow(<Input value={testString} />)
    const result = component.find('input').prop('value')

    expect(result).toEqual(testString)
  })

  it('Placeholder', () => {
    const testString = 'InputPlaceholder'
    const component = shallow(<Input placeholder={testString} />)
    const result = component.find('span.placeholder').text()

    expect(result).toEqual(testString)
  })

  it('Spy', () => {
    const event = { target: { name: 'search', value: 'qwerty' } }
    const component = shallow(<Input />)

    const spy = spyOn(component.instance(), 'onChange')
    component.instance().forceUpdate()
    component.find('input').simulate('change', event, spy)
    expect(spy.calls.mostRecent().args[0]).toEqual(event)
  })
})
