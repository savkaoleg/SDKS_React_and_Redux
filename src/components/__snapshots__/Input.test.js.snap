// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Renders correctly 1`] = `
<div
  className="form-input"
>
  <label>
    <input
      name="search"
      onChange={[Function]}
    />
    <span
      className="placeholder"
    />
  </label>
</div>
`;
